crypto/evp/evp_fetch.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stddef.h>
+#include <openssl/ossl_typ.h>
+#include <openssl/evp.h>
+#include <openssl/core.h>
+#include "internal/cryptlib.h"
+#include "internal/thread_once.h"
+#include "internal/asn1_int.h"
+#include "internal/property.h"
+#include "internal/core.h"
+#include "internal/evp_int.h"    /* evp_locl.h needs it */
+#include "evp_locl.h"
+
+/* The OpenSSL library context index for the default method store */
+static int default_method_store_index = -1;
+struct default_method_store_st {
+    OSSL_METHOD_STORE *store;
+    CRYPTO_RWLOCK *lock;
```

Comment:
The store contains a lock -- this seems like duplication.
I'd suggest leaving both in place for the moment and removing one later if it is possible.

Reply:
Oh right, I can just call the `ossl_method_store` functions without a surrounding lock.  I had forgotten

Reply:
So errr, contrary to your suggestion, I removed the lock here.

Reply:
No problem.  I wasn't sure if you'd need a lock at this level or not.  It seems not.

Reply:
No...  this code was much designed around earlier incantations of your store code, I think there wasn't a general lock back then.